### ISSUE TITLE
Rewrite main hunt view in React 

### ISSUE THREAD

**Issue by rgossiaux** [cardinalitypuzzles/smallboard#251]
Following on from #237 

You can view the progress locally by running `docker-compose up` and going to /hunts/react/[slug]. 

Copying over the task list from the old PR:

- [x] Real-time table updates
    - [x] Poll the puzzles API every 10s, as the current UI does.
    - [x] Make the collapse logic preserve collapse state across updates and not re-expand everything. Current smallboard doesn't do this; it's going to be complicated to get fully correct since we may need to map rows pre-update to rows post-update and that's not always going to be trivial: there may be duplicate rows (puzzles assigned to multiple metas) in different collapsed states, puzzles could be deleted or have their meta relationships change, etc, but we should be able to get the common cases correct at least.
- [x] Port functionality (UI + API)
    - [x] Adding a puzzle
    - [x] Deleting a puzzle
    - [x] Editing a puzzle
    - [x] Editing puzzle tags
    - [x] Meta assignment (@rgossiaux )
    - [x] Answer submission
    - [ ] Active puzzler display (probably deprecated)
-  [ ] Improve UI
    - [x] Style everything
    - [x] Render the correct table entries: tags, linkify URLs, remove extra debugging columns.
    - [ ] Display some basic stats (number of puzzles visible/total, num solved/pending/stuck/etc)
- [ ] Productionize
    - [x] Integrate webpack into the deployment pipeline
    - [x] Make the JS `prettier`
    - [x] Add testing, especially for the API
    - [ ] Update instructions and documentation

**Comment by rgossiaux** [comment]
Going to tackle some of the "improve UI" pieces next (especially rendering the right things in the table)

**Comment by rgossiaux** [comment]
Update: I implemented a first version of the delete puzzle modal/API locally (depends on #253) but the data flow is already kinda ugly. I'm refactoring it to use redux which should make the code a lot nicer.

I'm starting work on Monday so I'm going to try to get as much as I can done before then.

**Comment by rgossiaux** [comment]
I've linked a lot of things here so for convenience the current set of PRs is #253 => #256 => #257 => #258 .

It turns out my equipment for work won't arrive for at least a week so in the meantime I still have a lot of time to work on this. Hoping that by our meeting this Sunday we will almost be at the point where we could switch over. 

I'm going to work on error handling and possibly the webpack deployment aspects of this next. If anyone has free time over the next few days and wants to contribute, I think hooking up the "Assign Meta" and "Submit Answer" buttons should be pretty straightforward-- let me know!

**Comment by rgossiaux** [comment]
Latest update: on the features front, 

- I have a PR for tags at #269 
- @asdfryan has one for adding answers at #273 .
- I'm working on assigning metas locally (was busy over Christmas but have time for smallboard again the next few days) 
- we can probably either deprecate the active puzzler view or maybe port it over at a later point (either way I don't think it's critical). 
 
That covers all the features work.

Besides that, 
- @akirabaruah already added some documentation in #270 so I'll probably just expand on it slightly 
- we still need to write/port over tests to go through the API which shouldn't be that hard (probably parallelizable if someone with spare cycles in the next 1-2 days wants to help)

At that point I think we can cut over to the react UI by default and fully deprecate the old UI--should be able to do that before the new year. Then that leaves around two weeks for some forward progress on some related UI tasks (#68, #218, #220) which I think is doable. I'm also going to spend more time on testing & performance after "launching".

**Comment by rgossiaux** [comment]
Latest update: all the features are merged in and @asdfryan flipped the switch to use the React UI by default.

I didn't have as much time as I expected before the new year, but this weekend after UMD hunt I am going to add/update testing and documentation, and then I am going to close this issue and remove the old deprecated code.

**Comment by bcai2** [comment]
If you're writing the basic stats page/display, can you keep me in the loop? I'm thinking of adding the progress line chart (#200) to those once it's implemented just to keep everything stats-related in one place.

**Comment by rgossiaux** [comment]
Ok, with #305 we have some tests for the API and with #310 the old code is gone. Time to finally close this issue; anything UI related going forward can be tracked in its own issue.
